pkg/cmd/pulumi/new.go

```diff
@@ -246,49 +249,54 @@ func runNew(ctx context.Context, args newArgs) error {
 	fmt.Println()
 
 	// Load the project, update the name & description, remove the template section, and save it.
-	proj, root, err := readProject()
+	proj, path, err := readProjectWithPath()
+	root := filepath.Dir(path)
 	if err != nil {
 		return err
 	}
-	proj.Name = tokens.PackageName(args.name)
-	proj.Description = &args.description
-	proj.Template = nil
-	// Workaround for python, most of our templates don't specify a venv but we want to use one
-	if proj.Runtime.Name() == "python" {
-		// If the template does give virtualenv use it, else default to "venv"
-		if _, has := proj.Runtime.Options()["virtualenv"]; !has {
-			proj.Runtime.SetOption("virtualenv", "venv")
+
+	if filepath.Ext(path) == ".yaml" {
+		filedata, err := os.ReadFile(path)
+		if err != nil {
+			return err
+		}
+		var workspaceDocument yaml.Node
+		err = yaml.Unmarshal(filedata, &workspaceDocument)
+		if err != nil {
+			return err
 		}
-	}
 
-	if err = workspace.SaveProject(proj); err != nil {
-		return fmt.Errorf("saving project: %w", err)
-	}
+		err = yamlutil.Insert(&workspaceDocument, "name", args.name)
+		if err != nil {
+			return err
+		}
+		err = yamlutil.Insert(&workspaceDocument, "description", args.description)
+		if err != nil {
+			return err
+		}
+		err = yamlutil.Delete(&workspaceDocument, "template")
+		if err != nil {
+			return err
+		}
 
-	if proj.Runtime.Name() == "yaml" {
-		projFile := filepath.Join(root, "Pulumi.yaml")
-		f, err := ioutil.ReadFile(projFile)
+		projFile, err := yaml.Marshal(workspaceDocument.Content[0])
 		if err != nil {
-			return fmt.Errorf("could not find Pulumi.yaml: %w", err)
+			return err
 		}
-		appendFileName := "Pulumi.yaml.append"
-		appendFile := filepath.Join(root, appendFileName)
-		m, err := ioutil.ReadFile(appendFile)
-		if err == nil {
-			f = append(f, m...)
-			err = ioutil.WriteFile(projFile, f, 0600)
-			if err != nil {
-				return fmt.Errorf("failed to write %s: %w", projFile, err)
-			}
-			err = os.Remove(appendFile)
-			if err != nil {
-				return fmt.Errorf("could not remove %s: %w", appendFileName, err)
-			}
-		} else if !os.IsNotExist(err) {
-			return fmt.Errorf("could not get %s: %w", appendFileName, err)
+
+		err = os.WriteFile(path, projFile, 0600)
+		if err != nil {
+			return err
 		}
 	}
 
+	appendFileName := "Pulumi.yaml.append"
```

Comment:
This is just deleting the .append file? Doesn't the above need to read it and merge it in for correct yaml templates?

Reply:
We generated the .Append files by splitting the project file and taking the latter half, I believe @iwahbe had a script that did this.

See: 

https://github.com/pulumi/templates/blob/master/aws-yaml/Pulumi.yaml

We don't want to append the YAML file, we want to gracefully migrate away from needing the `.append` file in templates.

Reply:
Ah ok so the .yaml and .append.yaml both had all the resources in